### PR TITLE
[Improve-13001]migrate commons-collections -> commons-collections4

### LINKED ISSUE #13001: [Improvement][Migrate] Migrate org.apache.commons.collections -> org.apache.commons.collections4.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

+ The current DS uses both `commons-collections` and `commons-collections4`, we should migrate all `commons-collections` -> `commons-collections4`.  
+ Should update `commons-collections4` to `4.3`, because `commons-collections4 4.1`  is vulnerable
   > Please check: [Cx78f40514-81ff](https://devhub.checkmarx.com/cve-details/Cx78f40514-81ff/?utm_source=jetbrains&utm_medium=referral&utm_campaign=idea) 7.5 Uncontrolled Recursion vulnerability pending CVSS allocatio

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by DarkAssassinator**
## Purpose of the pull request

+ This PR will close #13001 

## Brief change log

+ Migrate `commons-collections` -> `commons-collections4`
+ Update commons-collections4 to 4.3 version to fix vulnerable
   > Please check: [Cx78f40514-81ff](https://devhub.checkmarx.com/cve-details/Cx78f40514-81ff/?utm_source=jetbrains&utm_medium=referral&utm_campaign=idea) 7.5 Uncontrolled Recursion vulnerability pending CVSS allocatio

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by kezhenxu94** [comment]
`commons-collections-3.2.2.jar` is still in dependency tree, can we totally remove this?

**Comment by DarkAssassinator** [comment]
> commons-collections-3.2.2.jar



> `commons-collections-3.2.2.jar` is still in dependency tree, can we totally remove this?

after i check dependency tree, we can found that many 3rd jars depend `commons-collections`, please check the following some result:
```
[INFO] org.apache.dolphinscheduler:dolphinscheduler-spi:jar:dev-SNAPSHOT
[INFO] \- org.apache.dolphinscheduler:dolphinscheduler-common:jar:dev-SNAPSHOT:compile
[INFO]    \- commons-beanutils:commons-beanutils:jar:1.9.4:compile
[INFO]       \- commons-collections:commons-collections:jar:3.2.2:compile

[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ dolphinscheduler-service ---
[INFO] Verbose not supported since maven-dependency-plugin 3.0
[INFO] org.apache.dolphinscheduler:dolphinscheduler-service:jar:dev-SNAPSHOT
[INFO] \- org.apache.hadoop:hadoop-common:jar:3.2.4:provided
[INFO]    \- commons-collections:commons-collections:jar:3.2.2:compile
```
And `commons-collections` and `commons-collections4` are diff artifactId, so we cannot remove `commons-collections-3.2.2.jar`.

File: dolphinscheduler-bom/pom.xml [hunk 1]
```diff
@@ -48,7 +48,7 @@
         <commons-beanutils.version>1.9.4</commons-beanutils.version>
         <commons-configuration.version>1.10</commons-configuration.version>
         <commons-email.version>1.5</commons-email.version>
-        <commons-collections4.version>4.1</commons-collections4.version>
+        <commons-collections4.version>4.3</commons-collections4.version>
```
Comment (ruanwenjun) [inline dolphinscheduler-bom/pom.xml:51]: Please update the LICENSE file.

Comment (DarkAssassinator) [reply dolphinscheduler-bom/pom.xml:51]: > Please update the LICENSE file.

done.

**Comment by kezhenxu94** [comment]
> > Please update the LICENSE file.
> 
> done.

This is not updated https://github.com/apache/dolphinscheduler/blob/dev/dolphinscheduler-dist/release-docs/LICENSE#L232

**Comment by DarkAssassinator** [comment]
> > > Please update the LICENSE file.
> > 
> > 
> > done.
> 
> This is not updated https://github.com/apache/dolphinscheduler/blob/dev/dolphinscheduler-dist/release-docs/LICENSE#L232

please check commit:https://github.com/apache/dolphinscheduler/pull/13002/commits/fe193b58a47804f571e9eff850850d7355109d4d

**Review by kezhenxu94 — DISMISSED** [review]

**Review by kezhenxu94 — APPROVED** [review]
LGTM, @ruanwenjun wanna recheck?

**Comment by DarkAssassinator** [comment]
@kezhenxu94 ,could help to merge?thx

**Pull request merged**